chore: add support for the openebs plugin

blaisedias opened the pull request:
the openebs plugin supercedes the mayastor
plugin.
However it is not a dropin replacement.
It is functional for all openebs IO engines,
and stem commands needs to further qualify
which IO engine the command is for.

At present invocations of the plugin in the
code base only exist for mayastor.

Refactor by adding functions
 -  GetMayastorPluginCmd returns and exec cmds with the
    correct argument list
 - GetMayastorPluginCmdString - needed for invocation of
   support shell scripts using the plugin
For the argument list is determined by name.
The assumption is that if the plugin name is
not kubectl-mayastor then it is the new plugin.

Refactor to replace most uses of GetPluginPath,
the exceptions being
  - upgrade
  - plugin version check commands

Also add support to create a config map for mayastor
diskpools.

Compatibilty with kubectl-mayastor is maintained

Commit:
chore: add support for the openebs plugin

the openebs plugin supercedes the mayastor
plugin.
However it is not a dropin replacement.
It is functional for all openebs IO engines,
and stem commands needs to further qualify
which IO engine the command is for.

At present invocations of the plugin in the
code base only exist for mayastor.

Refactor by adding functions
 -  GetMayastorPluginCmd returns and exec cmds with the
    correct argument list
 - GetMayastorPluginCmdString - needed for invocation of
   support shell scripts using the plugin
For the argument list is determined by name.
The assumption is that if the plugin name is
not kubectl-mayastor then it is the new plugin.

Refactor to replace most uses of GetPluginPath,
the exceptions being
  - upgrade
  - plugin version check commands

Also add support to create a config map for mayastor
diskpools.

Compatibilty with kubectl-mayastor is maintained

Signed-off-by: Blaise Dias <blaise.dias@datacore.com>

r1jt commented on common/k8stest/util_pool.go:
```diff
@@ -121,3 +130,131 @@ func GetCapacityAndAllocatedAllSnapshotsAllReplicasOnPool(poolName string) (int6
+						if weightI == weightJ {
+							// if weights are the same sort alphabetically for consitent
+							// results
+							return bd.DevLinks[i] < bd.DevLinks[j]
```
Return type is error . But here it is not?

blaisedias replied:
this is a return statement for the anonymous sorting function declared on line 206, so must be bool

r1jt commented on common/k8stest/util_pool.go:
```diff
@@ -121,3 +130,131 @@ func GetCapacityAndAllocatedAllSnapshotsAllReplicasOnPool(poolName string) (int6
+							// results
+							return bd.DevLinks[i] < bd.DevLinks[j]
+						}
+						return weightI < weightJ
```
Return type is error . But here it is not?

blaisedias replied:
this is a return statement for the anonymous sorting function declared on line 206, so must be bool

Commit:
chore: refactor: plugin invocation

Add support for specifying the full path
to the kubectl plugin.
Use that in preference over plugin dir
+ configured kubectl plugin name.

This gives invoking scripts more flexibility.

Signed-off-by: Blaise Dias <blaise.dias@datacore.com>

Commit:
chore: streamline: replace use of GetMayastorPluginCmdString

Replace use of GetMayastorPluginCmdString
with GetMayastorPluginCmd().String()

Signed-off-by: Blaise Dias <blaise.dias@datacore.com>
